core/modules/helpers/helpers.lua

```diff
@@ -0,0 +1,52 @@
+local set = require('set')
```

Comment:
Not 100% sure that a new helpers file would be the correct place for this.  What about building this into the existing DOM?

Reply:
I thought about putting it into the DOM but I wasn't sure how tightly we wanted to keep the DOM isolated from the various fields and stuff. Right now the dom files don't seem to operate on any of the fields so I didn't know if it'd be right to add this there. But I do agree it might be cleaner. Then you can just do `config:fetchAllIncludeDirs()` or `proj:fetchAllIncludeDirs()` without needing to use a new helpers module.

Reply:
Very true. Especially once we move to C# and such, having includeDirs functionality in DOM may not make ass much sense.